oitg/fitting/attenuated_sinusoid.py

```diff
@@ -0,0 +1,194 @@
+"""Fit an attenuated sinusoid with optional dead-time to data"""
+import numpy as np
+from scipy.signal import lombscargle
+from .FitBase import FitBase
+
+
+def fitting_function(t, p_dict):
+    """attenuated sinusoid with dead time"""
+    y = np.exp(-p_dict['rate'] * t) * \
+        (p_dict['a'] * np.sin((t - p_dict['t_dead']) * p_dict['omega']
+                              + p_dict['phi'])
+         + p_dict['c_offset']) + p_dict['c_equ']
+    # hold constant during dead time
+    return np.where(t > np.min(t) + p_dict['t_dead'], y,
+                    p_dict['c_offset'] + p_dict['c_equ'] +
+                    p_dict['a'] * np.sin(p_dict['phi'])
+                    )
+
+
+def init_all(t, y, p_dict):
+    """Initialise parameters under the general assumptions that:
+    t & y are 1D arrays of real numbers
+    associated data coordinates share the same index in t & y
+
+    Note: Data may be in any order and arbitrarily spaced"""
+    # p_dict is an instance of FitParameters
+    # user-inintialised values are not overwritten
+    # non user-initialised parameters are pre-initialised to 0
+
+    # sort data by time (x-axis)
+    mask = np.argsort(t)
+    t = t[mask]
+    y = y[mask]
+
+    p_dict['t_dead'] = 0.0
+    p_dict['rate'] = 0.0
+
+    min_step = np.min(t[1:] - t[:-1])
+    duration = t[-1] - t[0]
+    f_max = 0.5 / min_step
+    f_min = 0.5 / duration
+
+    omega_list = 2 * np.pi * np.linspace(f_min, f_max, int(f_max / f_min))
```

Comment:
Some comments would be nice. You could have copied those over from `rabi_flop`, where I am [also suggesting the frequency range heuristic you use](https://github.com/OxfordIonTrapGroup/oitg/blob/master/oitg/fitting/rabi_flop.py#L21-L26).

Reply:
I've improved the comments and documentation. Hope it's clear enough now.